deployer/src/persistence/mod.rs

```diff
@@ -466,6 +466,35 @@ impl ResourceManager for Persistence {
 
         Ok(res)
     }
+
+    async fn get_resource(
+        &self,
+        service_id: &Ulid,
+        r#type: ResourceType,
+        claim: Claim,
+    ) -> Result<Option<Resource>> {
+        sqlx::query_as(r#"SELECT * FROM resources WHERE service_id = ? AND type = ?"#)
+            .bind(service_id.to_string())
+            .bind(r#type)
+            .fetch_optional(&self.pool)
+            .await
+            .map_err(Error::from)
+    }
+
+    async fn delete_resource(
+        &self,
+        service_id: &Ulid,
+        r#type: ResourceType,
+        claim: Claim,
+    ) -> Result<()> {
+        sqlx::query(r#"DELETE FROM resources WHERE service_id = ? AND type = ?"#)
+            .bind(service_id.to_string())
+            .bind(r#type)
+            .execute(&self.pool)
+            .await
+            .map(|_| ())
+            .map_err(Error::from)
+    }
```

Comment:
We shouldn't need to delete from the sqlite database, resource deletion will only be available to new deployers, and they will use resource-recorder to record resources, not the sqlite database. So get_resource should use the resource-recorder client. Delete resource should first use the provisioner client to delete the resource, then if that succeeds it should use the resource-recorder client to delete the resource from the resource-recorder state.

Reply:
Hmm, we might also need to check that the resource is something the provisioner should delete. The provisioner only deals with databases currently. But we can probably start with the above comment and then tweak it after.

Reply:
I think this commit reflects some of your suggestions: https://github.com/shuttle-hq/shuttle/pull/1256#discussion_r1338706181.